### PR TITLE
Add ability to read CSV without header row

### LINKED ISSUE #54: CSV loader should support those without headers
Add an option for CSV files with no header

### PR DESCRIPTION
**Pull request by djalova**
---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #54

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [ ] When the pull request is ready, request review.
</details>

### CODE REVIEW

**Comment by djalova** [comment]
Recreating PR due to #81

File: pydax/loaders/_table.py [hunk 1]
```diff
@@ -37,6 +37,8 @@ def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDic
                - ``columns`` key specifies the data type of each column. Each data type corresponds to a Pandas'
                  supported dtype. If unspecified, then it is default.
                - ``delimiter`` key specifies the delimiter of the input CSV file.
+               - ``header`` key specifies if the first row of the CSV file contains the headers. Defaults to True
```
Comment (xuhdev) [inline pydax/loaders/_table.py]:
```suggestion
               - ``header`` key specifies if the first row of the CSV file contains the headers. Defaults to True.
```

File: tests/test_loaders.py [hunk 2]
```diff
@@ -243,3 +243,14 @@ def test_csv_pandas_loader_no_encoding(self, tmp_path, noaa_jfk_schema):
...
+    def test_csv_pandas_header(self, tmp_path, noaa_jfk_schema):
+        "Test CSVPandasLoader header options"
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['header'] = True
+        self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['header'] = False
+        with pytest.raises(ValueError) as exinfo:  # Pandas should error from trying to read string as another dtype
+            Dataset(noaa_jfk_schema, tmp_path, mode=Dataset.InitializationMode.DOWNLOAD_AND_LOAD)
+            assert('could not convert string to float' in exinfo.value)
```
Comment (xuhdev) [inline tests/test_loaders.py]: The exception is raised in the previous line and this assertion should have never been executed:

```suggestion
        assert 'could not convert string to float' in str(exinfo.value)
```

File: pydax/loaders/_table.py [hunk 3]
```diff
@@ -55,9 +57,18 @@ def load(self, path: Union[_typing.PathLike, Dict[str, str]], options: SchemaDic
             else:
                 dtypes[column] = type_
 
+        names = None
+        header = None
+        if options.get('header', True):
```
Comment (xuhdev) [inline pydax/loaders/_table.py]: Based on the document, do you mean

```suggestion
        if options.get('header', True) is not False:
```

Or you can actually make the function based on Python's evaluation of whether the value is true or false rather than deciding whether it is exactly `False` or not.

File: tests/test_loaders.py [hunk 4]
```diff
@@ -243,3 +243,14 @@ def test_csv_pandas_loader_no_encoding(self, tmp_path, noaa_jfk_schema):
...
         del noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['encoding']
         self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
+
+    def test_csv_pandas_header(self, tmp_path, noaa_jfk_schema):
+        "Test CSVPandasLoader header options"
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['header'] = True
+        self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['header'] = False
```
Comment (xuhdev) [inline tests/test_loaders.py]: Perhaps also add a test on the value being an empty string and `None`?

File: pydax/loaders/_table.py (hunk 3 above)
Comment (djalova) [reply pydax/loaders/_table.py]: I think it might be confusing if we make the function based on Python's evaluation. If we accidentally set `header` to `''`. Maybe we could rename the key to `no_header` and then use Python's evaluation?

File: tests/test_loaders.py (hunk 2 above)
Comment (djalova) [reply tests/test_loaders.py]: Whoops thanks for catching

File: pydax/loaders/_table.py (hunk 3 above)
Comment (xuhdev) [reply pydax/loaders/_table.py]: Sounds good.

File: tests/test_loaders.py [hunk 5]
```diff
@@ -243,3 +243,23 @@ def test_csv_pandas_loader_no_encoding(self, tmp_path, noaa_jfk_schema):
...
+        assert('could not convert string to float' in str(exinfo.value))
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['no_header'] = False
+        self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['no_header'] = ''
+        self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
+
+        noaa_jfk_schema['subdatasets']['jfk_weather_cleaned']['format']['options']['no_header'] = None
+        self.test_csv_pandas_loader(tmp_path, noaa_jfk_schema)
```
Comment (xuhdev) [inline tests/test_loaders.py]: How about using a for loop for these. Everything else LGTM now

**Review by xuhdev — APPROVED** [review]
LGTM
